shopify/product/update_when_out_of_stock/custom.js

```diff
@@ -0,0 +1,33 @@
+const Mesa = require('vendor/Mesa.js');
+
+/**
+ * A Mesa Script exports a class with a script() method.
+ */
+module.exports = new class {
+
+  /**
+   * Mesa Script
+   *
+   * @param {object} payload The payload data
+   * @param {object} context Additional context about this task
+   */
+  script = (payload, context) => {
+
+    let hasVariantsWithInventory = false;
+    
+    payload.variants.forEach(variant => {
+      // Check if variant quantities equal 0
+      if (variant.inventory_quantity > 0) {
+        hasVariantsWithInventory = true;
+      }
+    });
+
+    if (hasVariantsWithInventory === false) {
+      // We're done, call the next step!
+      Mesa.output.next(payload);
+    } else {
+      Mesa.log.error();
+      throw new Error('Variants have inventory. Unable to continue with next step.');
```

Comment:
Works great! But got several things:

1. Custom code throws an error when the product has inventory. I personally think it would be better if we just add some information in the Logs but keep the task as successful. Not sure if the error would be confusing for merchants who dont know what is going on in the workflow. They might wonder why its failing.

**My suggestion:**

 } else {
      Mesa.log.info("Variants have inventory. Unable to continue with next step.");
 }

2. If you wanna keep the product available on the online store, maybe get rid of this setting in the Shopify Update:
![Screen Shot 2021-06-17 at 11 42 50 AM](https://user-images.githubusercontent.com/42184178/122455388-30a79b00-cf61-11eb-9cd7-b03f1cad4b06.png)

Sort of defeats the purpose of editing the product description. 

3. Only adds the text " - OUT OF STOCK" to the first variant. My suggestion is to remove the last two steps. So that the merchant doesnt have to edit the variants text to remove "- OUT OF STOCK" when they restock the product. I think this is overkill. 
![Screen Shot 2021-06-17 at 11 44 17 AM](https://user-images.githubusercontent.com/42184178/122455544-5cc31c00-cf61-11eb-9985-cc1894097464.png)